Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -254,240 +168,272 @@ namespace Terrain
+                        AZ::Matrix3x4 matrix3x4 = AZ::Matrix3x4::CreateFromTransform(transform);
 
-                    resourceGroup->SetImage(m_heightmapImageIndex, m_areaData.m_heightmapImage);
-                    resourceGroup->SetConstant(m_modelToWorldIndex, matrix3x4);
+                        objectSrg->SetConstant(m_modelToWorldIndex, matrix3x4);
 
-                    ShaderTerrainData terrainDataForSrg;
-                    terrainDataForSrg.m_sampleSpacing = m_areaData.m_sampleSpacing;
-                    terrainDataForSrg.m_heightScale = m_areaData.m_heightScale;
-                    terrainDataForSrg.m_uvMin = uvMin;
-                    terrainDataForSrg.m_uvMax = uvMax;
-                    terrainDataForSrg.m_uvStep = uvStep;
-                    resourceGroup->SetConstant(m_terrainDataIndex, terrainDataForSrg);
+                        ShaderTerrainData terrainDataForSrg;
+                        terrainDataForSrg.m_sampleSpacing = m_areaData.m_sampleSpacing;
+                        terrainDataForSrg.m_heightScale = m_areaData.m_heightScale;
+                        terrainDataForSrg.m_uvMin = uvMin;
+                        terrainDataForSrg.m_uvMax = uvMax;
+                        terrainDataForSrg.m_uvStep = uvStep;
+                        objectSrg->SetConstant(m_terrainDataIndex, terrainDataForSrg);
 
-                    resourceGroup->Compile();
-                    drawPacketBuilder.AddShaderResourceGroup(resourceGroup->GetRHIShaderResourceGroup());
+                        objectSrg->Compile();
+                    }
 
-                    auto addDrawItem = [&](ShaderState& shaderState)
-                    {
-                        AZ::RHI::DrawPacketBuilder::DrawRequest drawRequest;
-                        drawRequest.m_listTag = shaderState.m_shader->GetDrawListTag();
-                        drawRequest.m_pipelineState = shaderState.m_pipelineState.get();
-                        drawRequest.m_streamBufferViews = AZStd::array_view<AZ::RHI::StreamBufferView>(&m_vertexBufferView, 1);
-                        drawPacketBuilder.AddDrawItem(drawRequest);
-                    };
-
-                    for (ShaderState& shaderState : m_shaderStates)
+                    m_sectorData.push_back();
+                    SectorData& sectorData = m_sectorData.back();
+
+                    for (auto& lod : m_patchModel->GetLods())
                     {
-                        addDrawItem(shaderState);
+                        AZ::RPI::ModelLod& modelLod = *lod.get();
+                        sectorData.m_drawPackets.emplace_back(modelLod, 0, m_materialInstance, objectSrg);
+                        AZ::RPI::MeshDrawPacket& drawPacket = sectorData.m_drawPackets.back();
+
+                        // set the shader option to select forward pass IBL specular if necessary
+                        if (!drawPacket.SetShaderOption(AZ::Name("o_meshUseForwardPassIBLSpecular"), AZ::RPI::ShaderOptionValue{ false }))
```

Comment:
Maybe put this AZ::Name as a member variable initialized once rather initializing each time in Render()